src/pint/models/timing_model.py

```diff
@@ -2639,6 +2697,269 @@ def __init__(self):
+        component_special_params = defaultdict(list)
+        for param, cps in self.param_component_map.items():
+            if len(cps) == 1:
+                component_special_params[cps[0]].append(param)
+        return component_special_params
+
+    def _add_alias_to_map(self, alias, param_name, alias_map):
+        """Add one alias to the alias-parameter map.
+        """
+        if alias in alias_map.keys():
+            if param_name == alias_map[alias]:
+                return alias_map
+            else:
+                raise ConflictAlias(
+                    f"Alias {alias} has been used by" f" parameter {param_name}."
+                )
+        else:
+            alias_map[alias] = param_name
+        return alias_map
+
+    def search_pulsar_system_components(self, system_name):
+        """Search the pulsar binary component based on given name.
+
+        Parameters
+        ----------
+        system_name : str
+            Searching name for the pulsar binary/system
+
+        Return
+        ------
+        The matching binary model component instance.
+
+        Raises
+        ------
+        UnknownBinaryModel
+            If the input binary model name does not match any PINT defined binary
+            model.
+        """
+        all_systems = self.category_component_map["pulsar_system"]
+        # Search the system name first
+        if system_name in all_systems:
+            return self.components[system_name]
+        else:  # search for the pulsar system aliases
+            for cp_name in all_systems:
+                if system_name == self.components[cp_name].binary_model_name:
+                    return self.components[cp_name]
+            raise UnknownBinaryModel(
+                f"Pulsar system/Binary model component"
+                f" {system_name} is not provided."
+            )
+
+    def alias_to_pint_param(self, alias):
+        """Translate a alias to a PINT parameter name.
+
+        This is a wrapper function over the property ``_param_alias_map``. It
+        also handles the indexed parameters (e.g., `pint.models.parameter.prefixParameter`
+        and `pint.models.parameter.maskParameter`) with and index beyond currently
+        initialized.
+
+        Parameters
+        ----------
+        alias : str
+            Alias name to be translated
+
+        Returns
+        -------
+        pint_par : str
+            PINT parameter name the given alias maps to. If there is no matching
+            PINT parameters, it will raise a `UnknownParameter` error.
+        first_init_par : str
+            The parameter name that is first initialized in a component. If the
+            paramere is non-indexable, it is the same as ``pint_par``, otherwrise
+            it returns the parameter with the first index. For example, the
+            ``first_init_par`` for 'T2EQUAD25' is 'EQUAD1'
+
+        Notes
+        -----
+        Providing a indexable parameter without the index attached, it returns
```

Comment:
What is the rationale for having it accept `F` and return `F0, F0` when `F7` would return `F7, F0`? Why accept un-numbered values at all?

Reply:
For JUMP, T2EQUAD...

Reply:
This is maybe changed, but it works for the current setup. We will change it when the maskParameter use the keyvalue as name